### ISSUE TITLE
`Gray` color model broken

### ISSUE THREAD

**Issue by SFr682k** [latex3/xcolor#23]
## Current behavior
Each of the following snippets fails with “Illegal unit of measure (pt inserted)” and “Missing number, treated as zero” errors.

### Snippet 1 ([Log file](https://github.com/latex3/xcolor/files/9295600/snippet-1.log)):
```latex
\documentclass{article}
\usepackage{xcolor}
\begin{document}
    \selectcolormodel{Gray}
    \definecolor{SomeColor}{Gray}{12}
    
    Hello World!
\end{document}
```

### Snippet 2 ([Log file](https://github.com/latex3/xcolor/files/9295601/snippet-2.log)):
```latex
\documentclass{article}
\usepackage{xcolor}
\begin{document}
    \selectcolormodel{Gray}
    \definecolor{SomeColor}{gray}{0.50}
    
    Hello World!
\end{document}
```

### Snippet 3 ([Log file](https://github.com/latex3/xcolor/files/9295633/snippet-3.log))
```latex
\documentclass{article}
\usepackage{xcolor}
\begin{document}
    \selectcolormodel{Gray}
    
    \color{red}
    Hello World!
\end{document}
```

## Expected behavior
The `Gray` color model “just works”

## Environment
 - TeX Live 2022
 - pdflatex 2022.6.25, pdftex v1.40.24
 - xcolor 2022/06/12 v2.14

**Comment by u-fischer** [comment]
yes. A quite old bug. Happily the hycolor package contains a patch (https://tex.stackexchange.com/a/520379/2388). I will move it to xcolor, but for now you can load  it directly (or load hyperref).

~~~~
\documentclass{article}
\usepackage{xcolor}
\usepackage{xcolor-patch}
\begin{document}
    \selectcolormodel{Gray}
    \definecolor{SomeColor}{gray}{0.50}

    Hello World!
\end{document}
~~~~

**Comment by u-fischer** [comment]
This has been fixed in version 3.00
